react/features/breakout-rooms/middleware.js

```diff
@@ -36,16 +38,48 @@ StateListenerRegistry.register(
 
 MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
     const { type } = action;
-    const result = next(action);
 
     switch (type) {
     case UPDATE_BREAKOUT_ROOMS: {
-        const { messages } = getState()['features/chat'];
+        // edit name if it was overwritten
+        const { overwrittenNameList } = getState()['features/base/participants'];
+
+        if (Object.keys(overwrittenNameList).length > 0) {
+            const newRooms = {};
+
+            Object.entries(action.rooms).forEach(([ key, r ]) => {
+                let participants = r?.participants || {};
+                let jid;
+
+                for (const id of Object.keys(overwrittenNameList)) {
+                    jid = Object.keys(participants).find(p => p.endsWith(id));
```

Comment:
This looks dangerous. Let's extract the resource ID (part after the last / ) and make sure we compare that, so we are sure it fully matches.